src/main/java/io/vertx/serviceproxy/InterceptorHolder.java

```diff
@@ -0,0 +1,39 @@
+package io.vertx.serviceproxy;
+
+import io.vertx.core.Future;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+
+import java.util.function.Function;
+
+public class InterceptorHolder {
+
+  private String action;
+
+  private Function<Message<JsonObject>, Future<Message<JsonObject>>> interceptor;
+
+  public InterceptorHolder(String action, Function<Message<JsonObject>, Future<Message<JsonObject>>> interceptor) {
+    this.action = action;
+    this.interceptor = interceptor;
+  }
+
+  public InterceptorHolder(Function<Message<JsonObject>, Future<Message<JsonObject>>> interceptor) {
+    this.interceptor = interceptor;
+  }
+
+  public String getAction() {
```

Comment:
Given that the action is now `final` this should be a "read-only" field so use should use the signature:

```
public String action()
```